Transform constraints to z3 constraints which is the final step

migeed-z opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #80744
* #80147
* __->__ #80110

Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

migeed-z committed:
Transform constraints to z3 constraints which is the final step

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"




[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"




[ghstack-poisoned]

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37387508).

migeed-z commented:
@migeed-z has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37387508).

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. 

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

jamesr66a commented on test/test_fx.py:
```diff
@@ -45,6 +45,7 @@
 
 if sys.version_info >= (3, 7):
     from fx.test_gradual_type import AnnotationsTest  # noqa: F401
+
```
Delete?

jamesr66a reviewed (approved):
This is a lot of code, but I like the test coverage so approving. I trust that this will be validated more in the future with the e2e cases

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z commented:
@pytorchbot merge -g

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z committed:
Update on "Transform constraints to z3 constraints which is the final step"


Differential Revision: [D37387508](https://our.internmc.facebook.com/intern/diff/D37387508)

We transform our constraints in a way that can be understood by z3. For example, for a dimension type, instead of Dyn, we encode it as a pair of integers. Some of the other operations are a 1:1 transformation.

contains tests for alexnet and resnet as well as smaller examples

[ghstack-poisoned]

migeed-z commented:
@pytorchbot merge -g
